### DISCUSSION TITLE
Hono lazy route?

### DISCUSSION THREAD

**Discussion by flamrdevs** [honojs/hono#2166] [Ideas]
It's possible to support dynamic import?

Example : 
```typescript
app.route("/path", () => import("./routes/path"));
```

**Comment by DianomiJH** [comment]
This would be _awesome_

**Reply by platform-kit** [reply]
@flamrdevs did you ever get anywhere with this?

**Reply by DianomiJH** [reply]
@platform-kit I ended up doing something like this:

```ts
app.get(
  "/foo/:id",
  async (context) => {
    const { handler } = await import(
      "./foo"
    );
    return handler(context);
  },
);
```

You miss out on type inference but you can fake it like this:

```ts
// foo.ts
import type { Context } from "hono";
import type { BlankEnv, BlankInput } from "hono/types";

export const handler = async (context: Context<BlankEnv, '/foo/:id', BlankInput>) => {...}
```

Doesn't work nicely with middleware but overall it's pretty good

**Comment by zipang** [comment]
I really support this feature request : it would be awesome to have a `lazy()` implementation for the react compatibility layer. It would make so much more libs work with `hono/jsx/dom` as a substitute for react !!
